docs/source/index.rst

```diff
@@ -9,15 +13,32 @@ On a single machine dask increases the scale of comfortable data from
 *fits-in-memory* to *fits-on-disk* by intelligently streaming data from disk
 and by leveraging all the cores of a modern CPU.
 
-Users interact with dask either by making graphs directly or through the *dask
-collections* which provide larger-than-memory counterparts to existing popular
-libraries:
+Dask use cases
+==============
+
+* **Scientific computing libraries:** Projects like `xray`_ and `scikit-image`_ 
+  use dask.array for fast interactive analysis of large numeric datasets.
+* **Large unstructured or semi-structured data:** Dask.bag excels at 
+  processing large JSON blobs and log files.
+* **Scaling pandas-like data analysis to large datasets:** Dask.dataframe 
+  implements a part of the pandas interface, and can process data files larger than 
+  available memory.
+* **Algorithm development:** Working directly with dask graphs or dask.imperative 
+  is an excellent way to implement and test new algorithms in fields such as linear 
+  algebra, optimization, and machine learning.
```

Comment:
You're right about starting from offhand comments and trying to evolve from there, and rethinking the section seems wise.

I think when a completely new user opens the dask docs, they'll be wondering what dask is, why it was created, and who uses it, how, and for what purpose. Use cases can help cover some of this ground, and my attempt here is to make the use cases a good way for a new user to first hear of dask.array, dask.bag, dask.dataframe, and dask.immediate.

Dask is for data analytics, which is inherently a very abstract and general goal, so it's a bit tricky finding ways to describe the use cases that aren't either so general that they're confusingly vague or so specific that they're misleadingly limiting. I think you have a great idea about discussing the general trend and then a specific example.

What would be a better version of this section that would talk about the purpose of dask and its collections, who might use them and how and why, with a reasonably concrete example or two for each one, and serve as a fairly brief friendly introduction to a user who is brand new to dask and as yet has no idea what it is or what it's for?